test/controllers/endorseCompanyClosureCertificate.controller.test.ts

```diff
@@ -56,4 +68,48 @@ describe('EndorseCompanyClosureCertificateController', () => {
       assert.isTrue(htmlAssertHelper.hasText('span#applicantName', 'John Smith'))
     })
   })
+
+  describe('POST - ensure form submission is handled correctly', () => {
+    it('should redirect successfully if validator returns no errors', async () => {
+      const testObject: EndorseCertificateFormModel = {confirmation: 'understood'}
+      const dissolutionPatchResponse: DissolutionPatchResponse = {
+        links: {
+          self: '',
+          payment: ''
+        }
+      }
+
+      when(mockedDissolutionService.approveDissolution(anything(), anything(), anything())).thenResolve(dissolutionPatchResponse)
+      when(mockedFormValidator.validate(deepEqual(testObject), formSchema)).thenReturn(null)
+
+      const app = createApp(container => {
+        container.rebind(FormValidator).toConstantValue(instance(mockedFormValidator))
+        container.rebind(SessionService).toConstantValue(instance(session))
+        container.rebind(DissolutionService).toConstantValue(instance(mockedDissolutionService))
+      })
+
+      await request(app).post(ENDORSE_COMPANY_CLOSURE_CERTIFICATE_URI)
+        .send(testObject)
+        .expect(MOVED_TEMPORARILY)
+        .expect('Location', PAYMENT_URI)
+    })
+  })
+
+  it('should render view with errors displayed if validator returns errors', async () => {
+    const testObject: EndorseCertificateFormModel = {confirmation: 'understood'}
+    const mockError: ValidationErrors = {
+      confirmation: `Test confirmation error`
+    }
+
+    when(mockedDissolutionService.approveDissolution(TOKEN, anything(), anything())).thenResolve()
+    when(mockedFormValidator.validate(deepEqual(testObject), formSchema)).thenReturn(mockError)
+
+    const app = createApp(container => {
+      container.rebind(FormValidator).toConstantValue(instance(mockedFormValidator))
+      container.rebind(SessionService).toConstantValue(instance(session))
+    })
+
+    const res = await request(app).post(ENDORSE_COMPANY_CLOSURE_CERTIFICATE_URI).send(testObject).expect(BAD_REQUEST)
+    assert.equal(res.text.match(/Test confirmation error/g)!.length, 2)
```

Comment:
this should use `htmlAssertHelper`. See more recent controllers for examples of how to assert that errors are displayed